pkg/util/admissioncheck/admissioncheck_test.go

```diff
@@ -0,0 +1,241 @@
+				t.Errorf("unexpected config (-want/+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestIndexerFunc(t *testing.T) {
+	cases := map[string]struct {
+		admissioncheck *kueue.AdmissionCheck
+		wantResult     []string
+	}{
+		"nil ac": {},
+		"wrong controller": {
+			admissioncheck: utiltesting.MakeAdmissionCheck("ac").
+				ControllerName("other-controller").
+				Parameters(kueue.GroupVersion.Group, "ProvisioningRequestConfig", "config-name").
+				Obj(),
+		},
+		"wrong ref": {
+			admissioncheck: utiltesting.MakeAdmissionCheck("ac").
+				ControllerName("test-controller").
+				Parameters(kueue.GroupVersion.Group, "NotProvisioningRequestConfig", "config-name").
+				Obj(),
+		},
+		"good": {
+			admissioncheck: utiltesting.MakeAdmissionCheck("ac").
+				ControllerName("test-controller").
+				Parameters(kueue.GroupVersion.Group, "ProvisioningRequestConfig", "config-name").
+				Obj(),
+			wantResult: []string{"config-name"},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			if err := clientgoscheme.AddToScheme(scheme); err != nil {
+				panic(err)
+			}
+			if err := kueue.AddToScheme(scheme); err != nil {
+				panic(err)
+			}
+
+			indexFnc := IndexerByConfigFunction("test-controller", kueue.GroupVersion.WithKind("ProvisioningRequestConfig"))
+
+			gotResult := indexFnc(tc.admissioncheck)
+
+			if diff := cmp.Diff(tc.wantResult, gotResult); diff != "" {
+				t.Errorf("unexpected result (-want/+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestFilterCheckStates(t *testing.T) {
+
+	cases := map[string]struct {
+		admissionchecks []kueue.AdmissionCheck
+		states          []kueue.AdmissionCheckState
+		wantResult      []string
+	}{
+		"empty": {},
+		"no match": {
+
+			states: []kueue.AdmissionCheckState{
+				{Name: "check1"},
+				{Name: "check2"},
+				{Name: "check3"},
+			},
+		},
+		"two matches": {
+			admissionchecks: []kueue.AdmissionCheck{
+				*utiltesting.MakeAdmissionCheck("check1").ControllerName("test-controller").Obj(),
+				*utiltesting.MakeAdmissionCheck("check2").ControllerName("other-controller").Obj(),
+				*utiltesting.MakeAdmissionCheck("check3").ControllerName("test-controller").Obj(),
+			},
+			states: []kueue.AdmissionCheckState{
+				{Name: "check1"},
+				{Name: "check2"},
+				{Name: "check3"},
+			},
+			wantResult: []string{"check1", "check3"},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			if err := clientgoscheme.AddToScheme(scheme); err != nil {
+				panic(err)
+			}
+			if err := kueue.AddToScheme(scheme); err != nil {
+				panic(err)
+			}
+			builder := fake.NewClientBuilder().WithScheme(scheme)
+			if len(tc.admissionchecks) > 0 {
+				builder = builder.WithLists(&kueue.AdmissionCheckList{Items: tc.admissionchecks})
+			}
+			client := builder.Build()
+			ctx := context.Background()
```

Comment:
I think we should make sure the context is closed in defer

Reply:
done